creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +436,139 @@ class SnapbackSM {
    * }
    */
   async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet, healthyNodes }) {
-    let currentNodeHasExistingUserState = true
-    let i = 0
-    let newSecondary
-    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
-      // Make sure current replica set will not be chosen as potential nodes for the new replica set
-      const healthyPotentialPeers = healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
+    let response = { newPrimary: null, newSecondary1: null, newSecondary2: null, issueReconfig: false }
+    // If entire replica set is unhealthy, select an entire new replica set
+    if (unhealthyReplicasSet.size === NUMBER_OF_REPLICA_SET_NODES) {
+      this.logWarn(`[determineNewReplicaSet] Entire replica set=[${Array.from(unhealthyReplicasSet)}] is unhealthy. Not issuing new replica set.`)
+      response.issueReconfig = this.isModeEnabled(reconfigModes.RECONFIG_DISABLED.key)
+      return response
+    }
+
+    const newReplicaNodes = await this.selectRandomReplicaSetNodes({
+      unhealthyReplicasSet,
+      healthyNodes,
+      wallet
+    })
+
+    // Perform last ditch effort to see if primary is truly unhealthy. If the primary responds
+    // with a 200 from the health check, remove from `unhealthyReplicaSet`. Else, continue with
+    // the new replica set determination logic.
+    const secondariesToClockMap = await this.fetchClockValues(secondary1, secondary2, wallet)
+    unhealthyReplicasSet = await this.finalizeUnhealthyReplicaSet({
+      unhealthyReplicasSet,
+      primary,
+      secondary1,
+      secondary2,
+      wallet,
+      secondariesToClockMap
+    })
+
+    // If the primary is still in the unhealthy replica set, and this primary for this wallet has not
```

Comment:
> it would be good to be more explicit that we will reconfig if our last check of this primary-user pair returned unhealthy

? is this not already in place with the current logic in this PR? if a primary for a wallet has been visited and marked as unhealthy in an earlier iteration, a reconfig should be issued

also, let's chat tomorrow via video about issuing reconfigs with relation to the current node